plugins/modules/purefa_pg.py

```diff
@@ -205,284 +213,394 @@
-        connected_arrays = get_arrays(array)
-        if OFFLOAD_API_VERSION in api_version:
-            connected_targets = get_targets(array)
-            offload_name = check_pg_on_offload(module, array)
-            if offload_name and offload_name in module.params["target"][0:4]:
-                module.fail_json(
-                    msg="Protection Group {0} already exists on offload target {1}.".format(
-                        module.params["name"], offload_name
-                    )
+        connected_arrays = get_arrays(module, array)
+        connected_targets = get_targets(module, array)
+        offload_name = check_pg_on_offload(module, array)
+        if offload_name and offload_name in module.params["target"][0:4]:
+            module.fail_json(
+                msg="Protection Group {0} already exists on offload target {1}.".format(
+                    module.params["name"], offload_name
                 )
+            )
 
         connected_arrays = connected_arrays + connected_targets
         if connected_arrays == []:
             module.fail_json(msg="No connected targets on source array.")
         if set(module.params["target"][0:4]).issubset(connected_arrays):
             if not module.check_mode:
-                try:
-                    array.create_pgroup(
-                        module.params["name"], targetlist=module.params["target"][0:4]
+                if LooseVersion(CONTEXT_API_VERSION) <= LooseVersion(api_version):
+                    res = array.post_protection_groups(
+                        names=[module.params["name"]],
+                        context_names=[module.params["context"]],
                     )
-                except Exception:
+                else:
+                    res = array.post_protection_groups(names=[module.params["name"]])
+                if res.status_code != 200:
                     module.fail_json(
-                        msg="Creation of replicated pgroup {0} failed. {1}".format(
-                            module.params["name"], module.params["target"][0:4]
+                        msg="Failed to create protection group {0}. Error: {1}".format(
+                            module.params["name"], res.errors[0].message
                         )
                     )
+                for target in range(0, len(module.params["target"][0:4])):
+                    if LooseVersion(CONTEXT_API_VERSION) <= LooseVersion(api_version):
+                        res = array.post_protection_groups_targets(
+                            group_names=[module.params["name"]],
+                            members=[FixedReference(name=target)],
+                            context_names=[module.params["context"]],
+                        )
+                    else:
+                        res = array.post_protection_groups_targets(
+                            group_names=[module.params["name"]],
+                            members=[FixedReference(name=target)],
```

Comment:
it should be members=[FixedReference(name=module.params["target"][target])], directly accessing target just gives 0.